scenario-compiler/src/main/java/pl/touk/nussknacker/engine/spel/internal/ConversionAndExtensionAwareMethodsDiscovery.java

```diff
@@ -0,0 +1,26 @@
+package pl.touk.nussknacker.engine.spel.internal;
+
+import pl.touk.nussknacker.engine.extension.Cast;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class ConversionAndExtensionAwareMethodsDiscovery {
+    private static final Method[] CAST_METHODS = Cast.class.getMethods();
+    private static final Method[] LIST_AND_CAST_METHODS = concatArrays(List.class.getMethods(), CAST_METHODS);
+
+    public Method[] discover(Class<?> type) {
```

Comment:
It can be. This class is extracted because in next PR it should be changed to automatically deliver newly registered extension methods. For now it's hardcoded that for array it should return list methods and for other just add cast methods to current methods. What I imagine is that the class will automatically add extension methods to the appropriate classes (I have part of this implementation locally but I will not make it public before my holidays).